qa/ccs-common-rest/src/yamlRestTest/java/org/elasticsearch/test/rest/yaml/CcsCommonYamlTestSuiteIT.java

```diff
@@ -306,4 +293,78 @@ public static void closeSearchClients() throws IOException {
+
+        TestCandidateAwareClient(
+            ClientYamlSuiteRestSpec restSpec,
+            RestClient restClient,
+            List<HttpHost> hosts,
+            Version esVersion,
+            Version masterVersion,
+            String os,
+            CheckedSupplier<RestClientBuilder, IOException> clientBuilderWithSniffedNodes
+        ) {
+            super(restSpec, restClient, hosts, esVersion, masterVersion, os, clientBuilderWithSniffedNodes);
+        }
+
+        public void setTestCandidate(ClientYamlTestCandidate testCandidate) {
+            this.testCandidate = testCandidate;
+        }
+
+        // we overwrite this method so the search client can modify the index names by prefixing them with the
+        // remote cluster name before sending the requests
+        public ClientYamlTestResponse callApi(
+            String apiName,
+            Map<String, String> params,
+            HttpEntity entity,
+            Map<String, String> headers,
+            NodeSelector nodeSelector
+        ) throws IOException {
+            // on request, we need to replace index specifications by prefixing the remote cluster
+            if (shouldReplaceIndexWithRemote(apiName)) {
+                String parameterName = "index";
+                if (apiName.equals("indices.resolve_index")) {
+                    // in this specific api, the index parameter is called "name"
+                    parameterName = "name";
+                }
+                String originalIndices = params.get(parameterName);
+                String expandedIndices = REMOTE_CLUSTER_NAME + ":*";
+                if (originalIndices != null && (originalIndices.isEmpty() == false)) {
+                    String[] indices = originalIndices.split(",");
+                    List<String> newIndices = new ArrayList<>();
+                    for (String indexName : indices) {
+                        newIndices.add(REMOTE_CLUSTER_NAME + ":" + indexName);
+                    }
+                    expandedIndices = String.join(",", newIndices);
+                }
+                params.put(parameterName, String.join(",", expandedIndices));
+            }
+            return super.callApi(apiName, params, entity, headers, nodeSelector);
+        }
+
+        private boolean shouldReplaceIndexWithRemote(String apiName) {
+            if (apiName.equals("scroll")
+                || apiName.equals("clear_scroll")
+                || apiName.equals("async_search.get")
+                || apiName.equals("async_search.delete")
+                || apiName.equals("async_search.status")
+                || apiName.equals("close_point_in_time")) {
+                return false;
+            }
+
+            if (apiName.equals("search") || apiName.equals("msearch") || apiName.equals("async_search.submit")) {
+                final String testCandidateTestPath = testCandidate.getTestPath();
+                if (testCandidateTestPath.equals("search/350_point_in_time/basic")
```

Comment:
Yes. The transformation here is to replace `index` with `remote:index`. But that does not work with PIT because it does not allow any indices to be specified. I added this block to not transform PIT tests so that they can run as part of this test suite.